internal/db/collection_define.go

```diff
@@ -271,7 +357,158 @@ func (db *DB) patchCollection(
 	return db.loadSchema(ctx)
 }
 
-// SetActiveSchemaVersion activates all collection versions with the given schema version, and deactivates all
+const (
+	collectionNamePathIndex int = 0
+	fieldsPathIndex         int = 1
+	fieldIndexPathIndex     int = 2
+)
+
+// substituteCollectionPatch handles any substitution of values that may be required before
+// the patch can be applied.
+//
+// For example Field [FieldKind] string representations will be replaced by the raw integer
+// value.
+func substituteCollectionPatch(
```

Comment:
note: This has been copy-pasted from the schema code, very few changes have been made - IIRC the only new feature is support for substitution in `"from"` paths, which had been missed in the original.